### DISCUSSION TITLE
Feature request - support for config file for Homebrew

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#968] [Everyday usage]
Homebrew supports reading config from env vars (like `HOMEBREW_GITHUB_API_TOKEN`) and from command line options (like `--force-bottle`), which is nice, but not very convenient IMO - 

Say a user wants core formula `python@3.9` to be always installed from the bottle, whether explicitly or as a dependency, he has to manually updates `python@3.9` first with `--force-bottle` before he can update any formulas that depends on `python@3.9` (or else it may be installed from source). 

With a config file, users may be able to 1) specify the formulas they wish to install from bottle / source 2) maintain a list of pinned formulas and 3) configure Homebrew with ease. Personally I think it would be a great feature for Homebrew to have. :)

**Comment by SMillerDev** [comment]
With a config file, users may be able to 

> 1) specify the formulas they wish to install from bottle / source

We only support bottle installs. Source builds are too prone to breakage.

> 2) maintain a list of pinned formulas 

This is not likely something we'll support since pinned formula are prone to breakage.

> 3) configure Homebrew with ease.

That's what we have brew bundle for.

**Reply by tddschn** [reply]
Thanks for the reply!

1. I do strongly prefer bottle installs, but some formulas like `python@3.9` seems to be always installed from source if not using `--force-bottle`, I'm not sure why.
2. I agree that pinning some core formulas might lead to breakage, but pinning formulas that is not a dependency for any other formulas like `elastic/tap/elasticsearch-full` is not likely to break things. 
And if formula A depends on pinned formula B, I don't think users can upgrade A without upgrading B first or using `--force`.
3. According to https://github.com/Homebrew/homebrew-bundle, brew bundle doesn't seem to support to configure brew the way env vars do. I use it as a package backup and restore tool, as it cannot configure homebrew itself.

**Reply by carlocab** [reply]
>some formulas like `python@3.9` seems to be always installed from source if not using `--force-bottle`, I'm not sure why

You're probably missing the CLT. Try installing it with `xcode-select --install`. (The CLT is different from Xcode.)
